ui/jstests/test_manage_taxonomies.jsx

```diff
@@ -91,4 +345,348 @@ define(['QUnit', 'jquery', 'setup_manage_taxonomies', 'reactaddons',
         );
     }
   );
+
+  QUnit.test('Assert that AddTermsComponent work properly',
+    function(assert) {
+      assert.ok(AddTermsComponent, "class object not found");
+      var vocabularies = [
+        {
+          "vocabulary": vocabulary,
+          "terms": vocabulary.terms
+        }
+      ];
+      var done = assert.async();
+      var addTermCalled = 0;
+      var afterMount = function(component) {
+        assertAddTermCommon(assert, component);
+        var inputGroup = React.addons.TestUtils.
+          findRenderedDOMComponentWithClass(
+            component,
+            'input-group'
+          );
+        var textbox = React.addons.TestUtils.
+          findRenderedDOMComponentWithTag(
+            inputGroup,
+            'input'
+          );
+        React.addons.TestUtils.Simulate.keyUp(textbox, {key: "Enter"});
+        waitForAjax(1, function() {
+          //test items
+          assert.equal(addTermCalled, 1);
+          done();
+        });
+      };
+      var addTerm = function() {
+        addTermCalled += 1;
+      };
+      var addTermsComponentRendered = React.addons.TestUtils.
+        renderIntoDocument(
+          <AddTermsComponent
+            vocabularies={vocabularies}
+            repoSlug="repo"
+            addTerm={addTerm}
+            ref={afterMount}
+          />
+        );
+      assert.equal(
+        addTermsComponentRendered.state.errorText,
+        ''
+      );
+      //test error message
+      addTermsComponentRendered.reportError(
+        "Error occurred while adding new term."
+      );
+      assert.equal(
+        addTermsComponentRendered.state.errorText,
+        'Error occurred while adding new term.'
+      );
+    }
+  );
+
+  QUnit.test('Assert that AddVocabulary work properly',
+    function(assert) {
+      assert.ok(AddVocabulary, "class object not found");
+      var vocabularies = [
+        {
+          "vocabulary": vocabulary,
+          "terms": vocabulary.terms
+        }
+      ];
+      var learningResourceTypes = ['course', 'chapter', 'sequential',
+        'vertical', 'html', 'video', 'discussion', 'problem'];
+      var saveVocabularyResponse;
+      var done = assert.async();
+      var updateParent = function(data) {
+        saveVocabularyResponse = data;
+      };
+      var afterMount = function(component) {
+        assert.equal(
+          component.state.name,
+          ''
+        );
+        assert.equal(
+          component.state.description,
+          ''
+        );
+        assert.equal(
+          component.state.vocabularyType,
+          'm'
+        );
+        assert.equal(
+          component.state.learningResourceTypes.length,
+          0
+        );
+        var formNode = assertAddVocabularyCommon(assert, component);
+        waitForAjax(1, function() {
+          assert.equal(
```

Comment:
Please add a comment indicating that `assertAddVocabularyCommon` submits the form and that you are checking that the state is cleared properly.